tests/testing_utils/fixtures.py

```diff
@@ -109,7 +109,8 @@ def __init__(self, rows: List[tuple], columns: List[str]):
             self._columns = [MockResultMetadata(c) for c in columns]
 
         def fetchone(self):
-            return self.fetchall()
+            if self._rows:
+                return self._rows.pop(0)
```

Comment:
We don't want handle else?

Reply:
Else we return None. I will add it explicitly